lang/java/reef-common/src/main/java/org/apache/reef/runtime/common/driver/evaluator/EvaluatorDescriptorImpl.java

```diff
@@ -52,15 +52,15 @@ public NodeDescriptor getNodeDescriptor() {
   }
 
   @Override
-  public synchronized EvaluatorType getType() {
-    return this.type;
+  public synchronized EvaluatorProcess getProcess() {
+    if (null == this.process) {
+      throw new IllegalArgumentException("EvaluatorProcess must be set");
+    }
+    return this.process;
   }
 
-  public synchronized void setType(final EvaluatorType type) {
-    if (this.getType() != EvaluatorType.UNDECIDED) {
-      throw new RuntimeException("Unable to change state of an Evaluator of Type: " + this.getType());
-    }
-    this.type = type;
+  public synchronized void setProcess(final EvaluatorProcess process) {
+    this.process = process;
```

Comment:
The process is now mutable. The default process is given in the constructor, and a user should be able to change it at any time.